generated/nifake/nifake/unit_tests/test_library_interpreter.py

```diff
@@ -410,6 +411,35 @@ def test_read_with_warning(self):
             assert test_error_desc in str(w[0].message)
             assert f'Warning {test_error_code} occurred.' in str(w[0].message)
 
+    def test_library_interpreter_always_uses_same_library_instance(self):
+        interpreter1 = nifake._library_interpreter.LibraryInterpreter('windows-1251')
+        interpreter2 = nifake._library_interpreter.LibraryInterpreter('windows-1251')
+        assert interpreter1 is not interpreter2
+        assert interpreter1._library is interpreter2._library
+
+    def test_set_runtime_environment_is_called_once_if_present(self):
+        nifake._library_interpreter._was_runtime_env_set = None
+        self.get_initialized_library_interpreter()
+        nifake._library_interpreter.LibraryInterpreter('windows-1251')
+        self.patched_library.niFake_SetRuntimeEnvironment.assert_called_once()
+
+    def test_set_runtime_environment_not_present_in_driver_runtime(self):
+        class TypesLibrary:
+            item = ""
+
+        nifake._library_interpreter._was_runtime_env_set = None
+        self.patched_library._library = TypesLibrary()
+        interpreter = self.get_initialized_library_interpreter()
+        with pytest.raises(nifake.errors.DriverTooOldError):
+            interpreter._library._get_library_function('niFake_SetRuntimeEnvironment')
+
+    def test_abort_not_present_in_library_raises_driver_too_old_error(self):
```

Comment:
Is this a valid test?
Under which circumstance would we have a function be present in the LibraryInterpreter that tries to call the Library, but the function isn't present in the library?


I wonder if we have coverage instead for the valid scenarion in which a new nimi-python module contains a function but the installed driver runtime is older and doesn't have the function.

Reply:
Yes @victorvadan is right. The above test does cover the case we care about. I think we can just delete this case.